File: ColorChangeGame/app/src/main/java/com/example/colorchangegame/MainActivity.java
package com.example.colorchangegame;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.os.LocaleListCompat;

import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

    Button button;

    Color colors;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Random random = new Random();
        List<Integer> colors = new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            int red = random.nextInt(256);
            int green = random.nextInt(256);
            int blue = random.nextInt(256);
            int color = Color.rgb(red, green, blue);
            colors.add(color);
        }

       button=(Button) findViewById(R.id.button);
       button.setOnClickListener(new View.OnClickListener() {
           @Override
           public void onClick(View v) {
               int randomColor = colors.get(new Random().nextInt(colors.size()));
               button.setBackgroundColor(randomColor);

           }
       });



    }







}

File: Nations/app/src/main/java/com/example/nations/MainActivity.kt
package com.example.nations

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    lateinit var recyclerView: RecyclerView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.recyclerView)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }


        recyclerView=findViewById(R.id.recyclerView)
        recyclerView.layoutManager= LinearLayoutManager(this, LinearLayoutManager.VERTICAL,false)
        recyclerView.adapter=Adapter(getModels())
    }

    fun getModels():MutableList<CountryModel>
    {
        val models= mutableListOf(
            CountryModel(R.drawable.usa,"ABD",
                "Washington",R.raw.abd),

            CountryModel(R.drawable.spain,"Ispanya",
                "Madrid",R.raw.spain),
            CountryModel(R.drawable.uk,"Birleşik Krallık",
                "Londra",R.raw.uk),

            CountryModel(R.drawable.serbia,"Sırbistan",
                "Belgrad",R.raw.serbia),

            CountryModel(R.drawable.turkey,"Türkiye",
                "Ankara",R.raw.turkey),

            CountryModel(R.drawable.argentina,"Arjantin",
                "Buenos Aires",R.raw.argentina),

            CountryModel(R.drawable.mexico,"Meksika",
                "Mexico City",R.raw.mexico),

            CountryModel(R.drawable.japan,"Japonya",
                "Tokyo",R.raw.japan),

            CountryModel(R.drawable.norway,"Norveç",
                "Oslo",R.raw.norway),

            CountryModel(R.drawable.portugal,"Portekiz",
                "Lizbon",R.raw.portugal)
        )
        return models
    }
}